### ISSUE TITLE
Anonymous records from a different assembly cannot be used with |>

### ISSUE THREAD

**Issue by alfonsogarciacaro** [dotnet/fsharp#7362]
When using an anonymous record in a signature of a function, if we call that function from a different project, we get the following error when instantiating the record and then using the pipe `|>` operator:

![image](https://user-images.githubusercontent.com/8275461/62617366-f8b98080-b911-11e9-8d8c-51689b60d870.png)

If we place the record after the function call then it works:

![image](https://user-images.githubusercontent.com/8275461/62617426-1f77b700-b912-11e9-8645-650db3eff31d.png)

**Comment by cartermp** [comment]
Repro:

Library code:

```fsharp
namespace lib

module Anon =
    let test (a: {| Intval: int; StringVal: string |}) = ()
```

Console code:

```fsharp
// Learn more about F# at http://fsharp.org

open System
open lib

let x = {| IntVal = 12; StringVal = "hello" |}
x |> Anon.test

[<EntryPoint>]
let main argv =
    printfn "Hello World from F#!"
    0 // return an integer exit code
```

In this case, calling `Anon.test x` also fails to compile.

**Comment by dsyme** [comment]
This is by design.  The two anon types are associated with different assemblies.   The assembly-affinity of anonymous types is covered in the RFC and is really a matter of "well, we-have-to-die-one-death" for this language feature.

**Comment by alfonsogarciacaro** [comment]
Thanks for your reply @dsyme. We discovered this in a big Fable solution where we have different apps and a shared project containing common components. For Fable the assembly boundary has no meaning as all sources are compiled as a single project at the end.

I mainly reported the issue because I noticed changing the order of the elements (placing the anonymous record **after** the function call) made the code compile (or at least the IDE stopped complaining) so I thought it'd be nice if the compiler could infer the anonymous record belonged to another assembly also when using `|>`. But if it's not possible I can live with it :)
